plugins/woocommerce/src/Blocks/AssetsController.php

```diff
@@ -62,9 +62,14 @@ public function register_assets() {
 		// The price package is shared externally so has no blocks prefix.
 		$this->api->register_script( 'wc-price-format', 'assets/client/blocks/price-format.js', array(), false );
 
-		$this->api->register_script( 'wc-blocks-vendors-frontend', $this->api->get_block_asset_build_path( 'wc-blocks-vendors-frontend' ), array(), false );
-		$this->api->register_script( 'wc-blocks-checkout', 'assets/client/blocks/blocks-checkout.js', array( 'wc-blocks-vendors-frontend' ) );
-		$this->api->register_script( 'wc-blocks-components', 'assets/client/blocks/blocks-components.js', array( 'wc-blocks-vendors-frontend' ) );
+		// Vendor scripts for blocks frontends (not including cart and checkout).
+		$this->api->register_script( 'wc-blocks-frontend-vendors', $this->api->get_block_asset_build_path( 'wc-blocks-frontend-vendors-frontend' ), array(), false );
+
+		// Cart and checkout frontend scripts.
+		$this->api->register_script( 'cart-checkout-vendors', $this->api->get_block_asset_build_path( 'cart-checkout-vendors-frontend' ), array(), false );
+		$this->api->register_script( 'cart-checkout-base', $this->api->get_block_asset_build_path( 'cart-checkout-base-frontend' ), array(), false );
+		$this->api->register_script( 'wc-blocks-checkout', 'assets/client/blocks/blocks-checkout.js' );
```

Comment:
We don't need to manually add dependencies any more because we add them to the asset.php files.